jina/executors/crafters/numeric/io.py

```diff
@@ -43,10 +42,10 @@ def craft(self, text: str, doc_id: int, *args, **kwargs) -> Dict:
             self.logger.error(
                 f'Data type mismatch. Cannot convert input to {self.as_type}.')
 
-        return dict(doc_id=doc_id, weight=1., blob=_array)
```

Comment:
@hanxiao I meant here for instance. How wil the indexer for instance know to which doc_id, 'now id of level 0' a particular item belongs?

Reply:
The principle, nested structure, `chunk_id` and `doc_id` are taken care by driver. Executor does not need to worry about these things.

- In general, `doc_id` should not be changed once assigned (often at client/request_gen).
- if you are doing crafting, you always work on the same doc, so there will be no change on `doc_id`. 
- when it comes to indexer, indexer stores & retrieves all documents in **flat**, not in a nested structure. well, at least I think it is not required. I will confirm it by refactoring the hello-world.

Reply:
Understood! Just trying to follow the main ideas. It is generally good that classes just care about the info they need to really perform their task, so good!